content/essentials/components-and-props.mdx

```diff
@@ -0,0 +1,339 @@
+
+### Destructuring props
+
+Props objects are readonly and have reactive properties.
+For this reason, when you destructure them you will lose reactivity if it is not done with a tracking scope, or if the reactive dependencies are not being tracked.
+
+```typescript
+function MyComponent(props) {
+  const name = props.name // ❌: breaks reactivity and will not update when the prop value changes
+  const [name, setName] = createSignal(props.name) // ✓: createSignal is a Solid primitive that track prop value changes
+}
+```
+
+In general, accessing properties on the props object outside of Solid's primitives or JSX can lose reactivity. 
+
+#### `splitProps`
+
+Solid provides the [`splitProps`](/reference/splitprops) utility for when destructuring is necessary. 
+This function is designed to help split a single props object into multiple sets of props, retaining the reactivity of the individual properties.
+
+The function allows you to define one or more arrays of keys that you wish to extract into separate props objects.
+It will return an array of props objects related to each set of keys, plus an additional props object containing any remaining keys. 
+Crucially, all the returned props objects will preserve their reactivity.
+
+This can be especially useful when you want to use certain props in the current component while passing others to child components:
+
+```typescript
+import { splitProps } from 'solid-js';
+
+function ParentComponent(props) {
+  // Splitting props into two groups: 'name' and 'age'
+  const [greetingProps, personalInfoProps, restProps] = splitProps(props, ['name'], ['age']);
+  
+  // Using greetingProps and personalInfoProps in the current component
+  return (
+    <div>
+      <Greeting {...greetingProps} />
+      <PersonalInfo {...personalInfoProps} />
+      {/* restProps can be passed down or used as needed */}
+    </div>
+  );
+}
+```
+### Passing props to children
+
+Solid's approach to performance revolves around ensuring components are lean and efficient. 
+In essence, Solid components can be visualized as mere function calls. 
+
+One of Solid's core strategies is using the compiler to wrap potentially reactive expressions within object [getters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get). 
+This allows these props to be evaluated [lazily](https://developer.mozilla.org/en-US/docs/Web/Performance/Lazy_loading), delaying their execution until their actual use.
+
+This approach promotes reactivity without the need for unnecessary [wrappers](https://developer.mozilla.org/en-US/docs/Glossary/Wrapper) or [synchronization](https://developer.mozilla.org/en-US/docs/Glossary/Synchronous). However, it has certain implications for handling child components. Due to this lazy evaluation, direct access to props can result in the repeated creation of child components or elements, which is preferable to sidestep.
```

Comment:
Technically, Solid compiles everything to unnecessary wrappers (also cool glossary link for wrapper). 

The synchronous glossary link looks incorrect to me

I think the lazy evaluation and repeated creation is a pretty advanced detail that should only be covered after advanced JSX compilation... Maybe just say reading props.children multiple times is unsafe and can cause problems like your code being slow?

Reply:
wrong in what way? it goes right to the MDN

Reply:
nvm i just removed it lol